### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by adrianmasiewicz**

### CODE REVIEW

**Review by Pawellos — APPROVED** [review]
Looks fine for me

File: src/functions.cpp [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-    }
-    std::cout << '\n';
-}
-
-void sort(std::vector<int> & unsorted) {
-    std::sort(begin(unsorted), end(unsorted));
-}
+#include "functions.hpp"
+ #include <iostream>
+ #include <algorithm>
```
Comment (Marcinati) [inline src/functions.cpp:3]: Sort headers.

File: src/functions.cpp [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-void print(std::vector<int> const & numbers) {
-    for (int i = 0; i < numbers.size(); ++i) {
-        std::cout << i << ": " << numbers[i] << '\n';
-    }
-    std::cout << '\n';
-}
-
-void sort(std::vector<int> & unsorted) {
-    std::sort(begin(unsorted), end(unsorted));
-}
```
Comment (Kasperice) [inline src/functions.cpp:15]: I was wondering why those lines are marked red (like deleted), then I saw that almost every line from the "green" (added) code has a whitespace in front. It is better to delete those whitespaces.

File: src/main.cpp [hunk 3]
```diff
@@ -1,14 +1,16 @@
-#include <vector>
```
Comment (Kasperice) [inline src/main.cpp:1]: Same comment here (whitespaces in front of line). In inserted code.

**Review by ziobron — CHANGES_REQUESTED** [review]
I can't see proper commit's structure. Have you been following the instruction and creating a commit after each step? I need to see a proper commit history here like below to grant you points. Can you try again? 
Zrzut ekranu 2020-06-7 o 17 17 23

File: src/main.cpp [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
-#include <iostream>
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+ #include <iostream>
+ #include <vector>
+ #include "functions.hpp"
```
Comment (DamiXioR) [inline src/main.cpp:3]: it's better when include header is on the top

**Review by DamiXioR — COMMENTED** [review]
Some ideas

File: src/main.cpp (hunk 4 above)
Comment (Marcinati) [reply src/main.cpp:3]: It's good according to Mateusz:

https://coders.school/edu/courses/powtorka-1-qa/lesson/kolejnosc-include/

**Comment by ziobron** [comment]
hey @adrianmasiewicz I just wanted to know if you need help with this one?

**Comment by ziobron** [comment]
I can grant you these points:
{ "requirement_ids": [1999, 2001] }
4 XP za Narzędzia#2 – vimtutor (1999)
4 XP za Narzędzia#2 – flagi kompilacji (2001)

Let me close this PR for now. If you wish to deliver changes please re-open it.

**Pull request closed without merging**